sim/src/simulator.rs

```diff
@@ -30,6 +66,20 @@ impl Simulator {
         self.elements[src].add_connection(dst.try_into().unwrap());
     }
 
+    pub fn print_graph(&mut self) {
```

Comment:
Graphviz is a fairly big dependency. Do we really need this in the simulator? What is it for?

Reply:
We don't strictly need it, but per the earlier comment I think it is a useful debugging tool and useful for people who want to use the simulator as a sanity check.

Reply:
Is there a way to use this without modifying the core code of the simulator? So keep it largely external, in a `utils` folder for example. Otherwise this may turn into feature creep. Ideally, the simulator code should be confined to minimal functionality.

Reply:
I think it has to be a part of the simulator to some extent;  the way it prints the graph is through making a graph as we create it in simulator.rs.  I could make it a command line option instead, so it doesn't happen every time and slow the code down with irrelevant things.  If you think it's feature creep I can also just get rid of it entirely.

Reply:
Petgraph includes dot already anyway so I guess we might as well make use of that. I would definitely make this a command line option though. And maybe just write a comment on what you need to install and what option to use, if you want to visualize the graph. Graphviz should  not be a default requirement.

Reply:
That sounds like a good plan.